packages/ra-ui-materialui/src/button/UpdateWithUndoButton.spec.tsx

```diff
@@ -95,11 +96,23 @@ describe('<UpdateWithUndoButton />', () => {
                     id: 123,
                     data: { views: 0 },
                     meta: undefined,
-                    previousData: { id: 123, title: 'lorem', views: 1000 },
+                    previousData: { id: 123, title: 'lorem', views: 500 },
                     resource: 'posts',
                 },
                 { snapshot: expect.any(Array) }
             );
         });
     });
+
+    it("should'nt open the show page caused by click propagation", async () => {
+        render(<WithAList />);
+        const startedUrl = global.window.location.pathname;
```

Comment:
I'm not 100% sure but I'd say this is bad practice to use `global.window.location.pathname` in tests. Please use text that only appear in the Show view for your tests instead.